src/App.tsx

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { getTodos, getUsers, Todo } from './helpers/api';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const handleLoadClick = async () => {
+    setIsLoading(true);
+
+    try {
+      const usersFromServer = await getUsers();
+      const todosFromServer = await getTodos();
+
+      const todosWithUsers = todosFromServer.map(todo => ({
+        ...todo,
+        user: usersFromServer.find(user => user.id === todo.userId),
+      }));
+
+      setTodos(todosWithUsers);
+      setIsLoaded(true);
+    } catch (error) {
+      setErrorMessage('Loading error, please try again later.');
+    }
+  };
+
+  const sortByTitle = () => {
+    const sortedTodods = [...todos].sort((a, b) => {
+      return a.title.localeCompare(b.title);
+    });
+
+    setTodos(sortedTodods);
```

Comment:
don't change initial todo list, each new sort action must use initial list

create a separate variable for sorted list, which will be calculated before rendering and passed to TodoList